View-Switching Navigation_Desktop/ViewSwitchingNavigation.Calendar/CalendarModule.cs

```diff
@@ -1,30 +1,43 @@
 
-
-using System.ComponentModel.Composition;
-using Prism.Mef.Modularity;
+using Microsoft.Practices.Unity;
 using Prism.Modularity;
 using Prism.Regions;
+using Prism.Unity;
+using ViewSwitchingNavigation.Calendar.Model;
 using ViewSwitchingNavigation.Calendar.Views;
+using ViewSwitchingNavigation.Calendar.ViewModels;
 using ViewSwitchingNavigation.Infrastructure;
 
 namespace ViewSwitchingNavigation.Calendar
 {
-    [ModuleExport(typeof(CalendarModule))]
     public class CalendarModule : IModule
     {
-        [Import]
-        public IRegionManager RegionManager;
+        private IUnityContainer unityContainer;
+        private IRegionManager regionManager;
+
+        public CalendarModule(IUnityContainer unityContainer, IRegionManager regionManager)
+        {
+            this.unityContainer = unityContainer;
+            this.regionManager = regionManager;
+        }
 
         public void Initialize()
         {
+            this.unityContainer.RegisterType<ICalendarService, CalendarService>(new ContainerControlledLifetimeManager());
+            this.unityContainer.RegisterType<CalendarViewModel>(new ContainerControlledLifetimeManager());
```

Comment:
Why are you registering Views and ViewModels with the container as singletons?  Don't do this.